src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -444,29 +455,33 @@ private TileEntityChest searchChestWithSimilarItem(final ItemStack stack)
      * @return the tileEntity of this chest.
      */
     @Nullable
-    private TileEntityChest searchMostEmptySlot()
+    private IItemHandler searchMostEmptySlot()
     {
         int freeSlots = 0;
-        TileEntityChest emptiestChest = null;
-        for(@NotNull final BlockPos pos : getBuilding().getAdditionalCountainers())
+        IItemHandler emptiestHandler = null;
+        for(@NotNull final BlockPos pos : getBuilding().getAdditionalContainers())
         {
             final TileEntity entity = worldObj.getTileEntity(pos);
             if(entity == null)
             {
                 getBuilding().removeContainerPosition(pos);
                 continue;
             }
-            if(entity instanceof TileEntityChest && InventoryUtils.getOpenSlot((IInventory) entity) != -1)
+            if(entity instanceof TileEntityChest)
             {
-                final int tempFreeSlots = ((TileEntityChest) entity).getSizeInventory() - InventoryUtils.getAmountOfStacks((IInventory) entity);
-                if(freeSlots < tempFreeSlots)
+                IItemHandler handler = entity.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'handler' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)